components/proxy/src/main/java/com/hotels/styx/StyxConfig.java

```diff
@@ -93,40 +74,30 @@ private static Configuration loadYamlConfiguration(String yaml) {
     }
 
     public StyxHeaderConfig styxHeaderConfig() {
-        return styxHeaderConfig;
+        return get("styxHeaders", StyxHeaderConfig.class).orElseGet(StyxHeaderConfig::new);
```

Comment:
Is this still needed?

Reply:
Not sure what you mean, but the change here is to simplify the way this object is constructed, which is still a valid refactoring.

Reply:
I don't see why we'd want to generate a new StyxHeaderConfig/Proxyconfig etc. object anytime we need to read it (that would be the case if the "configuration" object did not include the block). Isn't it simpler/preferable if it our StyxConfig object is immutable?

Reply:
These methods are rarely called - they are only needed at startup, so putting the parsing in the constructor was a premature optimisation.

Please note that these methods do not mutate StyxConfig in any way, so the class remains immutable.

Reply:
Of course, if you don't create fields in StyxConfig it is not mutable...  Should I have said I would rather have a snapshot of the whole config instead of having to load on demand? 

The "only needed at startup" assertion requires quite some code reading to verify. Specially in the case of ProxyToBackend.Factory.build() and StyxPipelineFactory.create(), and is probably contradicted by BackendServicesRouter.onChange() ... 

I don't think that a change that makes code simpler to reason about is an optimization but a good practice.

Reply:
I agree that it makes it easier to reason about. I will see if I can just revert that part.